### DISCUSSION TITLE
How to plot various infor in a way that is similar with IPF?

### DISCUSSION THREAD

**Discussion by Brian-Liu-rx** [mtex-toolbox/mtex#2012] [Ask Anything]
Hi, everyone,
My study is on the crystal plasticity of metal, and sometimes I want to make a plot which is much like the following picture. 
It can give the information very clearly that when loaded along various direction, what is the schmid factor of a certain slip system.
Certainly, the schmid factor is just taken as an example. I also want to plot other variables. 

![image](https://github.com/mtex-toolbox/mtex/assets/69452222/4e1a3ba1-e924-4d99-b4bf-903ec1e7f507)
Best regards,
Brian Liu

**Comment by kilir** [comment]
Hi,
in general the syntax would be `"plot(where,data)"`, so if you have a look at help orientation/plotIPDF e.g. `plotIPDF(ori,data,[r1,r2,r3])`.
Also, this has been asked quite a few times, check the search function here.
Cheers,
Rüdiger

**Reply by Brian-Liu-rx** [reply]
Hi, Rüdiger
Thanks for the quick reply!  I will try it!
BTW, I have asked many questions in the last year and you really have help me a lot. Thanks again!
Brian

**Reply by Brian-Liu-rx** [reply]
Hi, Rüdiger,
Now I can plot it! But I have another 2 issues now.
1. How can I add lables on the contourf lines?
2. As you can see, the contourf seems a little strange. There is a region with low value close to 001, which is not right. I guess this is due to the lack of data in this region. So how can I generate a very uniform orientation map or grid?
![image](https://github.com/mtex-toolbox/mtex/assets/69452222/c45bea8d-9551-4c2c-9a13-9cff4fee7791)

My code is :
orix = orientation.rand(2000,CS,SS);
[hkl,uvw] = round2Miller(orix); 
hkl = hkl.project2FundamentalRegion;
bbb = Miller(1,1,0,CS,'uvw');
nnn = Miller(1,-1,0,CS,'hkl');
allslips_110 = slipSystem(bbb,nnn).symmetrise('antipodal'); 
allslips_110_local = orix*allslips_110;
load_direction = -vector3d.Z;
all_SFs = zeros(6,2000);
for IS=1:6
    all_SFs(IS,:) = allslips_110_local(:,IS).SchmidFactor(load_direction);
end
all_SFs_max = max(abs(all_SFs))';
figure()
plot(hkl,all_SFs_max,'contourf','fundamentalRegion')
mtexColorbar


Thanks!
Brian

**Reply by Brian-Liu-rx** [reply]
I try this code and obtain a much better figure. This is good enough for me. But it is still nice to know how to plot a pretty contouf with lables 
figure()
plotIPDF(orix,all_SFs_max,vector3d.Z,'MarkerSize',5)
![Uploading image.png…]()

**Reply by Brian-Liu-rx** [reply]
@kilir

**Reply by kilir** [reply]
```
plot(hkl,all_SFs_max,'contourf','fundamentalRegion', ...
       'harmonic','bandwidth',64, ...
       'ShowText','on', ...
       'contours',[0:0.05:1])
mtexColorbar
```

**Reply by Brian-Liu-rx** [reply]
This is so great! Thanks a thousand times,Rüdiger.

**Reply by Brian-Liu-rx** [reply]
Rüdiger, the text is still not visible. @kilir 
![image](https://github.com/mtex-toolbox/mtex/assets/69452222/9643b58f-47fb-4f09-baeb-9be7ab655f0c)

**Reply by kilir** [reply]
Matlab refuses to label cluttered contours as well as contours that are too short. You may want to try to work with fewer contours.

**Reply by Brian-Liu-rx** [reply]
![image](https://github.com/mtex-toolbox/mtex/assets/69452222/8ed4e3e5-4796-4259-bb9e-847e2b2cc349)
Still not work,Rüdiger

**Reply by kilir** [reply]
Alternatively you can use `clabel`:
```
p  = plot(hkl,all_SFs_max,'contourf','fundamentalRegion','harmonic','bandwidth',64, ...
                          'contours',[0:0.05:1])
mtexColorbar

clabel(p.ContourMatrix, p)
```

Both produce this for me (using Matlab R2023b):

<img width="561" alt="contour" src="https://github.com/mtex-toolbox/mtex/assets/11697557/487c1576-52f4-4bcf-a42f-6ed6878c2a47">

**Reply by Brian-Liu-rx** [reply]
still no good. I guess something wrong with my Matlab. I will try it on another computer
